python/cudf/cudf/core/tokenize_vocabulary.py

```diff
@@ -0,0 +1,49 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+
+from __future__ import annotations
+
+import cudf
+from cudf._lib.nvtext.tokenize import (
+    Tokenize_Vocabulary as cpp_tokenize_vocabulary,
+    tokenize_with_vocabulary as cpp_tokenize_with_vocabulary,
+)
+
+
+class TokenizeVocabulary:
+    """
+    Parameters
+    ----------
+    vocabulary : str
+        Strings column of vocabulary terms
+    Returns
+    -------
+    TokenizeVocabulary
+    """
+
+    def __init__(self, vocabulary: "cudf.Series"):
+        self.vocabulary = cpp_tokenize_vocabulary(vocabulary._column)
+
+    def __call__(self, text, delimiter: str = "", default_id: int = -1):
```

Comment:
Typically Python libraries avoid using the `__call__` method for tasks like this, instead preferring a "verb" method. I propose to name this method `tokenize`, so that the user experience is more like:
```python
vocab = TokenizeVocabulary(words)
tokens = vocab.tokenize(more_words)
```

```suggestion
    def tokenize(self, text, delimiter: str = "", default_id: int = -1):
```